etcd/src/main/scala/com/github/fit51/reactiveconfig/etcd/ChannelManager.scala

```diff
@@ -41,12 +41,41 @@ object ChannelManager {
       authority: Option[String] = None,
       trustManagerFactory: Option[TrustManagerFactory] = None
   )(implicit exec: ExecutionContext, cl: Clock): ChannelManager with Authorization = {
-    val uris = endpoints.split(',').map(new URI(_)).toList
-    new ChannelManager(uris, options, authority, trustManagerFactory) with Authorization {
+    val uris = endpoints
+      .split(',')
+      .map(new URI(_))
+      .map(e => e.getHost() + (if (e.getPort() != -1) s":${e.getPort()}" else ""))
+      .mkString(",")
+    val target = s"https://${authority.getOrElse("etcd")}/$uris"
+    new ChannelManager(target, options, authority, trustManagerFactory) with Authorization {
       override val credentials    = credential
       override implicit val clock = cl
     }
   }
+
+  NameResolverRegistry
```

Comment:
I wonder, if this Resolver would be used in netty-code all over the project?
Wouldn't it affect other netty integrations?

Reply:
In that case, mb we should create a custom scheme (ex. etcd://) to isolate our custom registry?

Reply:
Have a look [here](https://github.com/grpc/grpc-java/issues/7133), guys are using the following kind of scheme `etcd:///authority/host1:port,host2:port`

Reply:
done